### PR TITLE
Omit calculation of derived_series() if Order is odd

### PR DESCRIPTION
**Pull request by divyanshu132**
#### Brief description of what is fixed or changed
[Feit–Thompson theorem](https://en.wikipedia.org/wiki/Feit%E2%80%93Thompson_theorem), or `odd order theorem`, states that every finite group of `odd order` is `solvable`.

#### Release Notes

*  combinatorics
   *  To check solvability omit calculation of derived_series() if Order is odd

### CODE REVIEW

**Comment by divyanshu132** [comment]
@jksuom please review.

File: sympy/combinatorics/perm_groups.py [hunk 1]
```diff
@@ -2027,6 +2027,8 @@ def is_solvable(self):
 
         """
         if self._is_solvable is None:
+            if self._order is not None and self._order % 2 != 0:
```
Comment (jksuom) [inline sympy/combinatorics/perm_groups.py]: Perhaps `self.order` could be used. That may take some time if it was not computed before, but computation of the derived series is also expensive.

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: I have tried that but it was showing `TypeError` somehow `self.order` behaves as a method.
 ```
   if self.order is not None and self.order % 2 != 0:
TypeError: unsupported operand type(s) for %: 'method' and 'int'

```

Comment (jksuom) [reply sympy/combinatorics/perm_groups.py]: `self.order` is a method, so `self.order() % 2 != 0` should be used. (In particular, it is never `None`.)

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: So sorry, I thought it's a property.

**Comment by divyanshu132** [comment]
@jksuom can this be merged.

**Comment by jksuom** [comment]
Can you find a test example of odd order?

**Comment by jksuom** [comment]
How about a non-commutative example? A Sylow subgroup of odd order of some S(n), for instance.

**Comment by divyanshu132** [comment]
@jksuom please review.

**Comment by jksuom** [comment]
Thanks.

**Pull request merged**